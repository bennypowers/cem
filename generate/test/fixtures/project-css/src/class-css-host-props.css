:host {
  /** defined on host */
  --defined-on-host: blue;

  /** also defined on host */
  --also-defined-on-host: blue;

  /** defined on host with var call */
  --defined-on-host-with-var: var(--defined-on-host);

  /** defined on host with light-dark call */
  --defined-on-host-with-var: light-dark(blue, red);
}

:host(.blue) {
  /** defined on host() */
  --defined-on-host-call: blue;

  /** also defined on host() */
  --also-defined-on-host-call: blue;
}

p {
  border:
    /** defined in var call */
    var(--defined-in-var-call)
    /** also defined in var call */
    var(--also-defined-in-var-call);
}
