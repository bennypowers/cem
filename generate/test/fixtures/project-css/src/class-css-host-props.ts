import styles from './class-css-host-props.css'
@customElement('class-css-host-props')
class ClassCssHostProps extends LitElement {
  static styles = styles;
}
