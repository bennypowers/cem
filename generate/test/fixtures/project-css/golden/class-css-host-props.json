{
  "schemaVersion": "2.1.0",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/class-css-host-props.js",
      "declarations": [
        {
          "name": "ClassCssHostProps",
          "superclass": {
            "name": "LitElement",
            "package": "lit"
          },
          "kind": "class",
          "tagName": "class-css-host-props",
          "cssProperties": [
            {
              "name": "--defined-on-host",
              "description": "defined on host",
              "default": "blue"
            },
            {
              "name": "--also-defined-on-host",
              "description": "also defined on host",
              "default": "blue"
            },
            {
              "name": "--defined-on-host-with-var",
              "description": "defined on host with var call",
              "default": "var(--expect-this-to-not-have-description)"
            },
            {
              "name": "--expect-this-to-not-have-description"
            },
            {
              "name": "--defined-on-host-with-light-dark",
              "description": "defined on host with light-dark call",
              "default": "light-dark(blue, red)"
            },
            {
              "name": "--defined-on-host-call",
              "description": "defined on host()",
              "default": "blue"
            },
            {
              "name": "--also-defined-on-host-call",
              "description": "also defined on host()",
              "default": "blue"
            },
            {
              "name": "--defined-in-var-call",
              "description": "defined in var call"
            },
            {
              "name": "--also-defined-in-var-call",
              "description": "also defined in var call"
            }
          ],
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "class-css-host-props",
          "declaration": {
            "name": "ClassCssHostProps",
            "module": "src/class-css-host-props.js"
          }
        }
      ]
    }
  ]
}
